mobile/src/main/java/ch/epfl/sdp/mobile/infrastructure/persistence/store/firestore/FirestoreQuery.kt

```diff
@@ -29,6 +29,12 @@ class FirestoreQuery(
   override fun whereArrayContains(field: String, value: Any): Query =
       FirestoreQuery(reference.whereArrayContains(field, value))
 
+  override fun startsWith(field: String, prefix: String): Query =
+      FirestoreQuery(
+          reference
+              .whereGreaterThanOrEqualTo(field, prefix)
+              .whereLessThan(field, prefix + '\uf8ff'))
```

Comment:
Yes, I should have added a comment on that. '\uf8ff' (The '<Private Use, Last>' Unicode character) is actually a very high code point in the Unicode range. Since it is after most regular characters, it can be used to query all string values starting with the given prefix since those values will fall within the provided range.